datafusion/physical-expr/src/equivalence/properties.rs

```diff
@@ -1677,14 +1677,20 @@ pub fn calculate_union(
 ) -> Result<EquivalenceProperties> {
     // TODO: In some cases, we should be able to preserve some equivalence
     //       classes. Add support for such cases.
-    let mut init = eqps[0].clone();
+    let mut iter = eqps.into_iter();
+    let Some(mut acc) = iter.next() else {
+        return internal_err!("Cannot calculate union of empty EquivalenceProperties");
```

Comment:
I'm not great expert here but if I caught this error I wouldn't have a slightest idea what is the reason tbh....

Reply:
```suggestion
        return internal_err!("Cannot calculate EquivalenceProperties for a union with no inputs");
```

Reply:
Hopefully this is a bit clearer